datafusion/physical-plan/src/joins/mod.rs

```diff
@@ -39,6 +40,11 @@ mod join_hash_map;
 #[cfg(test)]
 pub mod test_utils;
 
+/// The on clause of the join, as vector of (left, right) columns.
```

Comment:
It might make a nicer codebase if these were real structs rather than type defs 

```rust
pub struct JoinOn {
  inner: Vec<(PhysicalExprRef, PhysicalExprRef)>;
}
```

And then we could name the various accesses.

totally something for a follow on PR